File: frontend/src/app/interfaces/serie.ts
import { Actor, Cateogry, Language, Streaming } from "./movie";
import { ProfileInter } from "./profile";

export interface Series{
    id:number;
    title:string;
    description:string;
    duration:number;
    rating:number;
    age_rating:number;
    publication_year: number;
    episode:number;
    season:number;
    categoryIds:number[];
    streamingIds:number[];
    top10: boolean; 
    isInWatchlist: boolean;
}

export interface CreateSeries{
    title:string;
    description:string;
    synopsis:string;
    duration:number;
    rating:number;
    age_rating:number;
    publication_year: number;
    episode:number;
    season:number;
    categories: Cateogry[];   
    streamings: Streaming[];
    actors: Actor[];          
    languages: Language[];     
    top10: boolean;
}

export interface SeriesIndividual{
    id:number;
    title:string;
    description:string;
    synopsis:string;
    duration:number;
    rating:number;
    age_rating:number;
    publication_year: number;
    episode:number;
    season:number;
    categories: Cateogry[];
    streamings: Streaming[]; 
    top10:boolean;
    actors: Actor[];      
    languages: Language[];
    user:ProfileInter;
}

File: frontend/src/main/java/com/liaw/dev/cinemaflix/entity/Serie.java
package com.liaw.dev.cinemaflix.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.List;

@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_serie")
public class Serie {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    @Column(name = "synopsis", length = 4000)
    private String synopsis;
    private Integer rating;
    private Integer age_rating;
    private LocalDate publication_year;
    private Integer episode;
    private Integer season;
    private Boolean top10 = false;

    @ManyToMany
    @JoinTable(
            name = "serie_category",
            joinColumns = @JoinColumn(name = "serie_id"),
            inverseJoinColumns = @JoinColumn(name = "category_id")
    )
    private List<Category> categories;

    @ManyToMany
    @JoinTable(
            name = "serie_streaming",
            joinColumns = @JoinColumn(name = "serie_id"),
            inverseJoinColumns = @JoinColumn(name = "streaming_id")
    )
    private List<Streaming> streamings;

    @ManyToMany
    @JoinTable(
            name = "serie_actor",
            joinColumns = @JoinColumn(name = "serie_id"),
            inverseJoinColumns = @JoinColumn(name = "actor_id")
    )
    private List<Actor> actors;

    @ManyToMany
    @JoinTable(
            name = "serie_language",
            joinColumns = @JoinColumn(name = "serie_id"),
            inverseJoinColumns = @JoinColumn(name = "language_id")
    )
    private List<Language> languages;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

}
